### ISSUE TITLE
Is it possible to obtain the size of the shared folder?

### ISSUE THREAD

**Issue by avatar-lavventura** [owncloud/pyocclient#233]
Here I can obtain the shared file's information but I can't obtain its size. Is there any way to obtain its size?

```
#!/usr/bin/env python3

import owncloud

oc = owncloud.Client('https://b2drop.eudat.eu/')
oc.login('5f0db7e4-3078-4988-8fa5-f066984a8a97', '***********')

shareList = oc.list_open_remote_share()
# print(shareList)

for i in range(len(shareList)-1, -1, -1):
    inputFolderName  = shareList[i]['name']
    inputFolderName  = inputFolderName[1:] # Removes '/' on the beginning
    inputID          = shareList[i]['id']
    inputOwner       = shareList[i]['owner']

    if inputFolderName == '9a44346985f190a0af70a6ef6f0d35ee':
        print(shareList[i])
        print(inputOwner)
```

Example output of single shared file:

`{'id': '2609', 'remote': 'https://b2drop.eudat.eu/', 'remote_id': '12351', 'share_token': 'YUMcjKIZiSwEKy9', 'name': '/9a44346985f190a0af70a6ef6f0d35ee', 'owner': '059ab6ba-4030-48bb-b81b-12115f531296', 'user': '5f0db7e4-3078-4988-8fa5-f066984a8a97', 'mountpoint': '{{TemporaryMountPointName#/9a44346985f190a0af70a6ef6f0d35ee}}', 'accepted': '0'}`

**Comment by karakayasemi** [comment]
I am not familiar with pyoclient, but as far as I see in here  https://github.com/owncloud/pyocclient/blob/b9e1f04cdbde74588e86f1bebb6144571d82966c/owncloud/owncloud.py#L653 you list pending remote shares in your code.

If you want to see detailed info of a remote share, you should use this OCS_API: https://doc.owncloud.com/server/developer_manual/core/apis/ocs-share-api.html#get-information-about-a-known-federated-cloud-share, but I could not find pyoclient implementation of this API. You may try this method: https://github.com/owncloud/pyocclient/blob/b9e1f04cdbde74588e86f1bebb6144571d82966c/owncloud/owncloud.py#L868 , but I am not sure whether it works for pending remote share.

By the way, here is an issue tracking portal, https://central.owncloud.org is more suitable for questions. Thanks.

**Comment by avatar-lavventura** [comment]
You answer is not solve the issue.

=> `print(oc.get_share('2609'))` return `None`

There is `oc.file_info()` for file, which returns file's size in bytes but there is no option for folder.

**Comment by PVince81** [comment]
`file_info()` should also work on folders but might need to be modified to use the "oc:size" property instead of "getlastmodified"

if `get_share()` returns None it means the share is likely not found

**Comment by avatar-lavventura** [comment]
What do you mean by `need to be modified to use the "oc:size" property instead of "getlastmodified"`? Should I use `oc.size(f_name)` instead?

```
$ print(oc.file_info(f_name)
File(path=/ec2b1e7bd55e056dd34219d60b3a1422/ec2b1e7bd55e056dd34219d60b3a1422.tar.gz,file_type=file,attributes={'{DAV:}getlastmodified': 'Sun, 13 Oct 2019 09:15:31 GMT', '{DAV:}getcontentlength': '160', '{DAV:}resourcetype': None, '{DAV:}getetag': '"20cce987336efdd276a6057984c56722"', '{DAV:}getcontenttype': 'application/x-gzip'})

$ print(oc.file_info(f_name).attributes['{DAV:}getcontentlength'])
160
```

**Comment by PVince81** [comment]
no, the source code for `getsize()` needs to be adjusted here https://github.com/owncloud/pyocclient/blob/master/owncloud/owncloud.py#L247 to also take into account the Webdav property `{https://owncloud.org/ns}size`. It would also require extending https://github.com/owncloud/pyocclient/blob/master/owncloud/owncloud.py#L247 to actually query that property through PROPFIND.

I'll reopen this ticket as a feature request to extend this.

**Comment by avatar-lavventura** [comment]
Thanks @PVince81 .

What is the main difference between:
=> `oc.file_info(path_to_folder).attributes['{DAV:}getcontentlength']` and `getsize()`?

As I understand `file_info()` already works fine for folders, but for folder we have to get the size instead of getting it from `getlastmodified ` .

**Comment by PVince81** [comment]
@avatar-lavventura I wasn't aware of `.attributes`. in that case you could do `.attributes['{https://owncloud.org/ns}size']`.

however I believe that this property is not included by default in the PROPFIND response so it needs to be queried explicitly along with the other properties. For this, need to pass an XML body to the PROPFIND here https://github.com/owncloud/pyocclient/blob/master/owncloud/owncloud.py#L247 containing the desired properties.

**Comment by PVince81** [comment]
no, getcontentlength is not set for folders. you need the size property which is only set for folders

**Comment by PVince81** [comment]
the example you posted was for a zip file, try the same for a folder and you'll see that getcontentlength is not set for folders

**Comment by avatar-lavventura** [comment]
> the example you posted was for a zip file, try the same for a folder and you'll see that getcontentlength is not set for folders
You are right!

Sorry for confusion. For experimenting, I have a shared a folder which has multiple `.tar.gz` files and folders in it. And currently I am able to learn the size of the `tar.gz` files under the shared folder. 

On the end of shared account, it has to accept the shared folder using `oc.accept_remote_share(folder_shareID)`, if it does not accept it, it cannot retreive any information.

=> After accepting, I am able to get the size for the `.tar.gz` files, but not the parent folder, or folders inside the parent folder.

I can mount the `oc` directory, and after acccepting it accepted folder show up and I can learn its size, but it may require to download whole folder, I am not sure, or it may retrieve its size from its meta-file.

**Comment by DeepDiver1975** [comment]
This repository is being archived and is no longer maintained. Closing this as part of archival. Thank you for your contribution.
